### ISSUE TITLE
"Catching Rejected Promises" example

### ISSUE THREAD

**Issue by joshkel** [pburtchaell/redux-promise-middleware#89]
I'm afraid I may be missing something simple, but I'm having trouble applying the advice from ["Catching Rejected Promises"](https://github.com/pburtchaell/redux-promise-middleware/blob/master/docs/guides/rejected-promises.md).

From that file:

> create a second middleware, following the promise middleware, that globally catches all errors/rejected promises

What I was able to make work:

Create a second middleware, _before_ the promise middleware (because, otherwise, it doesn't see the promise middleware's return value, so it doesn't have access to the `Promise` object).

My middleware looks like this:

```js
export default function handleRejectedPromiseMiddleware(errorHandler) {
  return store => next => action => {
    let result = next(action);
    if (!errorHandler || !isPromise(result)) {
      return result;
    }

    return result.catch(errorHandler);
  };
}
```

(I'm borrowing redux-promise-middleware's [`isPromise`](https://github.com/pburtchaell/redux-promise-middleware/blob/master/src/isPromise.js) to implement this.)

Is this the correct approach?

I can submit a PR for rejected-promises.md with this or other example code if you like.

**Comment by pburtchaell** [comment]
@joshkel Yes, this implementation does sound correct. Let me verify this on one of my projects. I only wrote the current example as a sudo example and did not test it in actual code. I'll get back this weekend. 😄

**Comment by reintroducing** [comment]
I'm getting this error in IE11 only :\

Here is my middleware (as per @joshkel above, please do correct me if I'm doing something wrong, I'm extremely new to redux):
```
const isPromise = value => {
    if (value !== null && typeof value === 'object') {
        return value && typeof value.then === 'function';
    }

    return false;
};

export default function handleRejectedPromiseMiddleware(errorHandler) {
    return store => next => action => {
        const result = next(action);

        if (!errorHandler || !isPromise(result)) {
            return result;
        }

        return result.catch(errorHandler);
    };
}
```

and my store:
```
export const store = createStore(
    reducers,
    initialState,
    compose(
        applyMiddleware(
            routingMiddleware,
            thunkMiddleware,
            handleRejectedPromiseMiddleware(),
            promiseMiddleware()
        ),
        window.devToolsExtension ? window.devToolsExtension() : f => { return f; }
    )
);
```

In IE11: https://www.dropbox.com/s/f0267y06rqa1t49/Screenshot%202016-07-21%2021.23.10.png?dl=0

Appreciate any help you guys can offer.

**Comment by reintroducing** [comment]
Forgot to add what my actions look like, here is an example:
```js
import {get} from 'utils/api';

export const LOAD_MESSAGES = 'LOAD_MESSAGES';

export const fetchMessages = ({
    userId,
    conversationId,
    filter = 'all',
    limit = -1,
    page = 1
}) => {
    return {
        type: LOAD_MESSAGES,
        payload: get(`user/${userId}/messages`, {conversationId})
    };
};
```

**Comment by reintroducing** [comment]
Aaaaand completely disregard my prior posts, I don't even need the middleware, this was happening because of something completely unrelated and I've now fixed it and I'm all good.

Apologies for the noise :\

**Comment by pburtchaell** [comment]
Hey @joshkel, your issue is correct. Would you mind making a PR to update the docs? Otherwise, I will make the update this week. 👍

**Comment by pburtchaell** [comment]
I also updated the complex example with [a global error middleware](https://github.com/pburtchaell/redux-promise-middleware/blob/update-example/examples/complex/middleware/error.js). This will be merged into master once #105 is merged. 😄

**Comment by pburtchaell** [comment]
Actually, @joshkel, never mind. Tonight, [I started new documentation](https://github.com/pburtchaell/redux-promise-middleware/tree/documentation) for the version 4 release. I'll just include an updated guide on this when I make those updates.

**Comment by joshkel** [comment]
Thanks for the feedback and confirmation.

**Comment by DeDuckProject** [comment]
Hi @pburtchaell,
I'm using redux-promise-middleware and the rejected promises crash.
I tried using the global error middleware (copied the code from this repo).
I get a result.catch is not a function.
<img width="391" alt="screenshot 2016-08-17 17 46 39" src="https://cloud.githubusercontent.com/assets/10045034/17740937/1e424f24-64a3-11e6-8b6c-65bb8cca0428.png">

This happens, as I understand, because the promises I'm using are not ES6 promises, but regular promises that don't have 'catch'.

Is there a way to totally prevent the middleware to throw errors? 
I want errors to be handled by me in the reducer.

big thanks!
